starter-code/main.js

```diff
@@ -0,0 +1,390 @@
+
+                        db.collection('companies').find({}, { name: 1, _id: 0 }).toArray((error, result) => {
+                            if (error) {
+                                console.log(error);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            } else {
+                                console.log(result);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            }
+                        })
+                        break;
+
+                    case "2":
+
+                        db.collection('companies').find({}, { name: 1, _id: 0 }).toArray((error, result) => {
+                            if (error) {
+                                console.log(error);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            } else {
+                                console.log(result.length);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            }
+                        })
+
+                        break;
+
+                    case "3":
+
+                        db.collection('companies').find({ founded_year: 2004 }, { name: 1, _id: 0 }).toArray((error, result) => {
+                            if (error) {
+                                console.log(error);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            } else {
+                                console.log(result.length);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            }
+                        })
+
+                        break;
+
+                    case "4":
+
+                        db.collection('companies').find({ $and: [{ founded_year: 2004 }, { founded_month: 2 }] }, { name: 1, _id: 0 }).toArray((error, result) => {
+                            if (error) {
+                                console.log(error);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            } else {
+                                console.log(result);
+                                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                            }
+                        })
+
+                        break;
+
+                    case "5":
+
+                        db.collection('companies').find(
+                            {
+                                $and: [{ founded_year: 2004 },
+                                {
+                                    $and: [{ founded_month: { $gte: 4 } },
+                                    { founded_month: { $lte: 6 } }]
+                                }]
```

Comment:
solo un comentario, la decision de utilizar la conjunción lógica AND es la correcta, sin embargo podría emplearse simplemente de la siguiente manera:

```
   .find(
                {
                  $and: [
                    { founded_month: { $gte: 4 } },
                    { founded_month: { $lte: 6 } },
                    { founded_year: 2004 },
                  ],
                },
```